core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -169,6 +170,9 @@ public void validateIndexName(String index, ClusterState state) {
         if (index.equals(".") || index.equals("..")) {
             throw new InvalidIndexNameException(new Index(index), index, "must not be '.' or '..'");
         }
+        if (index.startsWith(DUMMY_INDEX_PREFIX)) {
+            throw new InvalidIndexNameException(new Index(index), index, "must not start with '" + DUMMY_INDEX_PREFIX + "'");
+        }
```

Comment:
Can we avoid reserving a prefix and instead generate a unique index name, maybe using something like `UUIDs.randomBase64UUID()`?